packages/@ember/-internals/owner/index.ts

```diff
@@ -16,25 +19,26 @@ export interface Factory<T, C extends FactoryClass | object = FactoryClass> {
   create(props?: { [prop: string]: any }): T;
 }
 
-export interface EngineInstanceOptions {
-  mountPoint: string;
-  routable: boolean;
-}
-
-import EngineInstance from '@ember/engine/instance';
-import { TypeOptions } from '../container/lib/registry';
+// A combination of the public methods on ContainerProxyMixin and RegistryProxyMixin
```

Comment:
As a next step (doesn't need to be in this PR, but is related to #19916), we should add docs for all of these bits of public API.

That in turn will let us expose a copy of these types in DefinitelyTyped, which will unblock folks between now and when we publish these.